indico/modules/events/registration/util.py

```diff
@@ -406,6 +406,9 @@ def create_registration(regform, data, invitation=None, management=False, notify
             continue
         default = form_item.field_impl.default_value
         can_modify = management or not form_item.parent.is_manager_only
+        if (invitation and invitation.lock_email and form_item.type == RegistrationFormItemType.field_pd and
+                form_item.personal_data_type == PersonalDataType.email):
+            can_modify = False
```

Comment:
This does not work as expected, you end up creating registrations with a blank email (which I'm surprised we don't prohibit with a DB constraints!).

```py
if ...:
    value = invitation.email
elif can_modifiy:
    value = data.get(form_item.html_field_name, default)
else:
    value = default
```